frontend/src/views/courses/roles/StudentCourseView.vue

```diff
@@ -47,28 +43,28 @@ async function leaveCourse(): Promise<void> {
     confirm.require({
         message: t('confirmations.leaveCourse'),
         header: t('views.courses.leave'),
-        accept: (): void => {
+        accept: async (): Promise<void> => {
             if (user.value !== null) {
                 // Leave the course
-                studentLeaveCourse(props.course.id, user.value.id).then(() => {
-                    // Refresh the user so the course is removed from the user's courses
-                    refreshUser();
-                    // Redirect to the dashboard
-                    push({ name: 'dashboard' });
-                });
+                await studentLeaveCourse(props.course.id, user.value.id);

```

Comment:
Shouldn't this be an emit too?

Reply:
I guess it does not matter because it gets redirected to the dashboard anyway

Reply:
> Shouldn't this be an emit too?

It's a view, so no